sdk/firstload/src/index.tsx

```diff
@@ -380,25 +381,25 @@ const H: HighlightPublicInterface = {
 	): any => {
 		const tracer = typeof getTracer === 'function' ? getTracer() : undefined
 		if (!tracer) {
+			console.log('::: no tracer')
+			const noopSpan = getNoopSpan()
+
 			if (fn === undefined && context === undefined) {
-				;(options as Callback)()
+				return (options as Callback)(noopSpan)
 			} else if (fn === undefined) {
-				;(context as Callback)()
+				return (context as Callback)(noopSpan)
 			} else {
-				fn()
+				return fn(noopSpan)
 			}
-			return
 		}
 
-		const wrapCallback = async (
-			span: Span,
-			callback: (span: Span) => any,
-		) => {
+		const wrapCallback = (span: Span, callback: (span: Span) => any) => {
+			let result: any
 			try {
-				const result = await callback(span)
-				return result
+				result = callback(span)
```

Comment:
if the callback is a promise, does that mean line 401 (`span.end()`) will happen before the promise is finished? seems like a regression in expected behavior

Reply:
Ah, you're right! Thanks for catching that. I think there was also a bug with our types for this function which should have said it is returning a promise.

Reply:
Just addressed this and added a test for async callbacks in https://github.com/highlight/highlight/pull/9335/commits/3099b9fe042aa312e84dbf41b3e6c3ecfdf8e46f